python/pycylon/data/table.pyx

```diff
@@ -259,156 +384,194 @@ cdef class Table:
+        """
+        creating a PyCylon table from a list
+        @param context: CylonContext
+        @param col_names: Column names
+        @param data_list: data as a list
+        @return: PyCylon table
+        """
+        ar_list = []
+        if len(col_names) == len(data_list):
+            for data in data_list:
+                ar_list.append(data)
+            return Table.from_arrow(context, pa.Table.from_arrays(ar_list, names=col_names))
+        else:
+            raise ValueError("Column Names count doesn't match data columns count")
+
+    @staticmethod
+    def from_pydict(context: CylonContext, dictionary: dict) -> Table:
         """
-        [Experimental]
-        This method converts a Cylon Table to a 2D numpy array.
-        In the conversion we stack each column in the Table and create a numpy array.
-        For Heterogeneous Tables, use the generated array with Caution.
-        :param order: numpy array order. 'F': Fortran Style F_Contiguous or 'C' C Style C_Contiguous
-        :return: ndarray
+        creating a PyCylon table from a dictionary
+        @param context: CylonContext
+        @param dictionary: dict with table data
+        @return: PyCylon table
         """
-        table = to_pyarrow_table(self.id.encode())
-        py_arrow_table = pyarrow_wrap_table(table)
+        return Table.from_arrow(context, pa.Table.from_pydict(dictionary))
+
+    @staticmethod
+    def from_pandas(context: CylonContext = None, df: pd.DataFrame = None, preserve_index=False,
+                    nthreads=None, columns=None, safe=False) -> Table:
+        """
+            creating a PyCylon table from Pandas DataFrame
+            :param obj: Pandas DataFrame
+            :rtype: PyCylon Table
+        """
+        return Table.from_arrow(context,
+                                pa.Table.from_pandas(df=df, schema=None,
+                                                     preserve_index=preserve_index,
+                                                     nthreads=nthreads, columns=columns, safe=safe)
+                                )
+
+    def to_pandas(self):
+        """
+         creating Pandas Dataframe from PyCylon Table
+         :param self:
+         :return: a Pandas DataFrame
+         """
+        return self.to_arrow().to_pandas()
+
+    def to_numpy(self, order: str = 'F'):
+        """
+         [Experimental]
+         This method converts a Cylon Table to a 2D numpy array.
+         In the conversion we stack each column in the Table and create a numpy array.
+         For Heterogeneous Tables, use the generated array with Caution.
+         :param order: numpy array order. 'F': Fortran Style F_Contiguous or 'C' C Style C_Contiguous
+         :return: ndarray
+         """
         ar_lst = []
         _dtype = None
-        for col in py_arrow_table.columns:
+        for col in self.to_arrow().columns:
             npr = col.chunks[0].to_numpy()
```

Comment:
shall we do a merge chunks operation here, before converting to np?

Reply:
why?

Reply:
it could be a chunked array and if we only convert the 0 chunk, we'll loose data, right?

Reply:
could you elaborate a bit?

Reply:
I think what Niranda tries to say is, if the table has chunked arrays (read from csv methods etc) this will result in we missing the rest of the data in the other chunks

Reply:
I see, so once a table is generated after an RA op or just by loading. Does it have multiple chunks? 
Or is this like a validation?

Reply:
It can have multiple chunks

Reply:
well, we can't guarantee that the underlying table only has a single chunk in each of its columns. You can use this method. 
http://arrow.apache.org/docs/python/generated/pyarrow.Table.html?highlight=combine%20chunks#pyarrow.Table.combine_chunks

Reply:
```for col in self.to_arrow().combine_chunks().columns:
            npr = col.chunks[0].to_numpy()
```
This should work?

Reply:
yep. that should work